### PR TITLE
Ebpf xdp update 5.0 v6

### PR DESCRIPTION
**Pull request by regit**
Update of #4037 addressing comments

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Describe changes:
- Address comments in doc and code

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/487
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/265

### CODE REVIEW

File: doc/userguide/capture-hardware/ebpf-xdp.rst [hunk 1]
```diff
@@ -64,12 +71,17 @@ Kernel
...
 
 Make sure you have clang (>=3.9) installed on the system  ::
 
- sudo apt-get install clang
+ sudo apt install clang
+
+Some i386 headers will also be needed as eBPF is not x86_64 and some included headers
+are architecture specific ::
+
+ sudo apt install  libc6-dev-i386 --no-install-recommends
```
Comment (norg) [inline doc/userguide/capture-hardware/ebpf-xdp.rst]: double spaces between install and libc6.

Might also be helpful to link so some explanation why it's not x86_64 and if there is a way for 64-Bit only systems that don't have 32-bit libs anymore.

File: doc/userguide/capture-hardware/ebpf-xdp.rst [hunk 2]
```diff
@@ -403,9 +445,12 @@ configuration of this interface ::
   - interface: eth3
     pinned-maps: true
 
+XDP and pinned-maps
+-------------------
+
 This option can be used to expose the maps of a socket filter to other processes.
 This allows for example, the external handling of a accept list or block list of
-IP addresses. See `scbpf` tool avalable in the `ebpf/scpbf` directory for an example
+IP addresses. See `bpfctrl <https://github.com/StamusNetworks/bpfctrl/>`__ for an example
```
Comment (norg) [inline doc/userguide/capture-hardware/ebpf-xdp.rst]: are those __ after the link necessary?

File: ebpf/xdp_lb.c [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+static int __always_inline filter_ipv4(struct xdp_md *ctx, void *data, __u64 nh_off, void *data_end)
+{
+    struct iphdr *iph = data + nh_off;
+    if ((void *)(iph + 1) > data_end)
+        return XDP_PASS;
+
+    if (iph->protocol == IPPROTO_GRE) {
+        __be16 proto;
```
Comment (victorjulien) [inline ebpf/xdp_lb.c]: could this be moved into a subroutine? Every path in it leads to a return.

File: doc/userguide/capture-hardware/ebpf-xdp.rst [hunk 4]
```diff
@@ -54,7 +60,8 @@ This guide has been confirmed on Debian/Ubuntu "LTS" Linux.
 Disable irqbalance
 ~~~~~~~~~~~~~~~~~~
 
-::
+Irqbalance may cause issue in most setups described here, so it is recommended
```
Comment (victorjulien) [inline doc/userguide/capture-hardware/ebpf-xdp.rst]: issue**s**

File: ebpf/filter.c [hunk 5]
```diff
@@ -72,6 +80,34 @@ int SEC("filter") hashfilter(struct __sk_buff *skb) {
     return -1;
 }
 
+static __always_inline int ipv6_filter(struct __sk_buff *skb)
+{
+    return -1;
+}
+
+int SEC("filter") hashfilter(struct __sk_buff *skb) {
```
Comment (victorjulien) [inline ebpf/filter.c]: Newline before curly bracket

File: ebpf/xdp_lb.c [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+#define KBUILD_MODNAME "foo"
```
Comment (victorjulien) [inline ebpf/xdp_lb.c]: what is this for?

File: doc/userguide/capture-hardware/ebpf-xdp.rst (hunk 2 above)
Comment (regit) [reply doc/userguide/capture-hardware/ebpf-xdp.rst]: one only is needed.

File: ebpf/xdp_lb.c (hunk 6 above)
Comment (regit) [reply ebpf/xdp_lb.c]: It was needed at one time to get the eBPF installed into the kernel. It is still present in some eBPF kernel sample. I don't know if we could have side effect  of removing it with some old kernel.

**Pull request closed without merging**